components/tidb_query_expr/src/impl_time.rs

```diff
@@ -553,6 +553,33 @@ pub fn add_duration_and_duration(
     Ok(Some(res))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn add_duration_and_string(
+    ctx: &mut EvalContext,
+    arg1: &Duration,
+    arg2: BytesRef,
+) -> Result<Option<Duration>> {
+    let arg2 = std::str::from_utf8(&arg2).map_err(Error::Encoding)?;
+    let arg2 = match Duration::parse(ctx, arg2, MAX_FSP) {
```

Comment:
string cloud be datetime here?

Reply:
Please take https://github.com/tikv/tikv/pull/8192 a look.

Reply:
i see it in [non-vec framework before it was removed](https://github.com/tikv/tikv/blob/1a88f12ebf50064b992fc5efc7e7f55795210521/components/tidb_query_normal_expr/src/builtin_time.rs)  435 line that is datetime.

Reply:
Ohh, I got it.